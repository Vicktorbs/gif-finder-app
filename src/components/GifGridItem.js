import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({ title, url }) => {
    return (
        <a href={ url } target='_blank' rel='noreferrer' className="card card-background">
            <div className="animate__animated animate__fadeIn">
                <img src={ url } alt={ title } />
                <p>{ title }</p>
            </div>
        </a>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}