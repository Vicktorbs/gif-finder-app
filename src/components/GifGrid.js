import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
    // key: sgatASouL6QA0OloiTwxS5gAVHGXDGET
    // const [images, setImages] = useState([])
    const { data:images, loading } = useFetchGifs(category);

    return (
        <>
            <li className="cards-container" key={ category }>
                <h3 className="animate__animated animate__fadeIn card-title">{ category }</h3>
                { loading && <p>Loading...</p> }
                <div className="card-grid">
                    { 
                        images.map(image => 
                            <GifGridItem { ...image } key={ image.id } />
                        )
                    }
                </div>
            </li>
        </>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}