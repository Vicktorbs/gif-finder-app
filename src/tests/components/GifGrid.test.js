import { shallow } from 'enzyme';
import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test on File GifGrid', () => {
    
    test('should show <GifGrid /> correctly', () => {
        // REgresa informacion "falsa" que se podra utilizar cuando se implemente el custom hook
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={ 'hinata' }/>);
        expect(wrapper).toMatchSnapshot();
    })

    test('should show items when useFetchGifs loads images', () => {
        const gifs = [{
            id: 'qKr1nIRfIQPYI',
            url: 'https://media1.giphy.com/media/qKr1nIRfIQPYI/giphy.gif?cid=c4f74b61bdxuxtgjry3au947bwjp4fc6uncmdmfvt0b48g7v&rid=giphy.gif&ct=g',
            title: 'i just love her naruto shippuden GIF'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={ 'hinata' }/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
})
