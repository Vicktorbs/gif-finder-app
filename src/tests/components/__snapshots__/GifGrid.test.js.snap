// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on File GifGrid should show <GifGrid /> correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    hinata
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test on File GifGrid should show items when useFetchGifs loads images 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    hinata
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="qKr1nIRfIQPYI"
      key="qKr1nIRfIQPYI"
      title="i just love her naruto shippuden GIF"
      url="https://media1.giphy.com/media/qKr1nIRfIQPYI/giphy.gif?cid=c4f74b61bdxuxtgjry3au947bwjp4fc6uncmdmfvt0b48g7v&rid=giphy.gif&ct=g"
    />
  </div>
</Fragment>
`;
